### PR TITLE
exhibitor updates

### PR DESCRIPTION
**Pull request by alberts**

### CODE REVIEW

File: gen/dcos-config.yaml [hunk 1]
```diff
@@ -284,17 +284,27 @@ package:
...
       OAUTH_CLIENT_ID={{ oauth_client_id }}
       OAUTH_AUTH_REDIRECTOR={{ oauth_auth_redirector }}
+  - path: /etc/lb_dnsnames
+    content: |
+      MASTER_INTERNAL_LB_DNSNAME={{ exhibitor_address }}
 {% switch master_discovery %}
 {% case "master_http_loadbalancer" %}
+  - path: /etc_master/dns_config_master
+    content: |
+      MASTER_SOURCE=exhibitor_uri
```
Comment (cmaloney) [inline gen/dcos-config.yaml]: Does spartan even know what to do with this?

Comment (alberts) [inline gen/dcos-config.yaml]: I think we're good.

https://github.com/dcos/spartan/blob/master/src/spartan_config_loader_server.erl#L114

File: gen/installer/aws.py [hunk 2]
```diff
@@ -233,6 +233,7 @@ def make_advanced_bunch(variant_args, template_name, cc_params):
         extra_templates=extra_templates,
         cc_package_files=[
             '/etc/cfn_signal_metadata',
+            '/etc/lb_dnsnames',
```
Comment (cmaloney) [inline gen/installer/aws.py]: this file doesn't exist

File: packages/mesos/build [hunk 3]
```diff
@@ -32,37 +33,37 @@ cp /usr/lib/x86_64-linux-gnu/libsvn_delta-1.so.1 "$libdir"
...
 
 export PKG_PATH=$PKG_PATH
 systemd_master="$PKG_PATH"/dcos.target.wants_master/dcos-mesos-master.service
 mkdir -p "$(dirname "$systemd_master")"
 envsubst '$PKG_PATH' < /pkg/extra/dcos-mesos-master.service > "$systemd_master"
 
-systemd_slave="$PKG_PATH"/dcos.target.wants_slave/dcos-mesos-slave.service
-mkdir -p "$(dirname "$systemd_slave")"
-envsubst '$PKG_PATH' < /pkg/extra/dcos-mesos-slave.service > "$systemd_slave"
+systemd_agent="$PKG_PATH"/dcos.target.wants_agent/dcos-mesos-agent.service
```
Comment (cmaloney) [inline packages/mesos/build]: _slave and _slave_public aren't change to _agent and _agent_public currently. Way to late for that.

File: packages/mesos/extra/dcos-mesos-agent.service [hunk 4]
```diff
@@ -14,7 +14,9 @@ EnvironmentFile=-/opt/mesosphere/etc/mesos-slave-common-extras
 EnvironmentFile=/opt/mesosphere/etc/mesos-slave
 EnvironmentFile=-/var/lib/dcos/mesos-slave-common
 EnvironmentFile=/var/lib/dcos/mesos-resources
+EnvironmentFile=-/run/dcos/etc/mesos-slave
```
Comment (cmaloney) [inline packages/mesos/extra/dcos-mesos-agent.service]: Can we please do `/run/dcos/<service-name>/<whatever the service wants>`? This current layout makes making all of this stuff non-root next to impossible without moving everything.

File: packages/mesos/build [hunk 5]
```diff
@@ -32,37 +33,37 @@ cp /usr/lib/x86_64-linux-gnu/libsvn_delta-1.so.1 "$libdir"
...
 disk_resource_script="$PKG_PATH/bin/make_disk_resources.py"
 cp /pkg/extra/make_disk_resources.py "$disk_resource_script"
 chmod +x "$disk_resource_script"
 
-# Copy the json file describing logrotate module.
-module_json="$PKG_PATH/etc/mesos-slave-modules/logrotate.json"
-mkdir -p "$(dirname "$module_json")"
-cp /pkg/extra/logrotate.json "$module_json"
+# Install agent modules.
+mkdir -p "$PKG_PATH/etc/mesos-agent-modules"
```
Comment (cmaloney) [inline packages/mesos/build]: keep this slave-modules for now. Should switch all the mesos config at once, not half of it here.

**Comment by alberts** [comment]
This PR adds some debug logging for Spartan.

https://mesosphere.atlassian.net/browse/DCOS-8790 will remind us to remove it.

**Comment by cmaloney** [comment]
Spartan in AWS CF is 100% broken with this. Reverting.

```Jul 21 14:04:18 ip-10-0-0-255.us-west-2.compute.internal systemd[1]: Starting DNS Dispatcher: An RFC5625 Compliant DNS Forwarder...
Jul 21 14:04:19 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Clearing proxy environment variables
Jul 21 14:04:19 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] master_count file doesn't exist, not waiting
Jul 21 14:04:19 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Connecting to leader.mesos:2181
Jul 21 14:04:19 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [WARNING] Connection dropped: socket connection error: Name or service not known
Jul 21 14:04:20 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Connecting to leader.mesos:2181
Jul 21 14:04:20 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [WARNING] Connection dropped: socket connection error: Name or service not known
Jul 21 14:04:20 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Connecting to leader.mesos:2181
Jul 21 14:04:20 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [WARNING] Connection dropped: socket connection error: Name or service not known
Jul 21 14:04:21 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Connecting to leader.mesos:2181
Jul 21 14:04:21 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [WARNING] Connection dropped: socket connection error: Name or service not known
Jul 21 14:04:21 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Connecting to leader.mesos:2181
Jul 21 14:04:21 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [WARNING] Connection dropped: socket connection error: Name or service not known
Jul 21 14:04:21 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Connecting to leader.mesos:2181
Jul 21 14:04:21 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [WARNING] Connection dropped: socket connection error: Name or service not known
Jul 21 14:04:22 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Connecting to leader.mesos:2181
Jul 21 14:04:22 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [WARNING] Connection dropped: socket connection error: Name or service not known
Jul 21 14:04:22 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Connecting to leader.mesos:2181
Jul 21 14:04:22 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [WARNING] Connection dropped: socket connection error: Name or service not known
Jul 21 14:04:22 ip-10-0-0-255.us-west-2.compute.internal bootstrap[15901]: [INFO] Connecting to leader.mesos:2181
```

**Pull request merged**